### ISSUE TITLE
Segmentation fault on Manjaro with mesa-git RADV

### ISSUE THREAD

**Issue by Krteq** [SaschaWillems/VulkanCapsViewer#142]
Hi, whatever I'm trying to do I'm still getting SIGSEGV with vulkanCapsViewer 3.21 on Manjaro Linux

Tried to build AUR package, AppImage and last attempt was a built from git source.

Here is a GDB trace for last attempt:

```
(gdb) run ./vulkanCapsViewer
Starting program: /home/krteq/Install/VulkanCapsViewer/vulkanCapsViewer ./vulkanCapsViewer
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[New Thread 0x7ffff2a7c640 (LWP 7339)]
[New Thread 0x7fffebddd640 (LWP 7340)]
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
[New Thread 0x7fffd2dff640 (LWP 7341)]
Reading Vulkan 1.1 core properties
Reading Vulkan 1.2 core properties
Reading Vulkan 1.3 core properties
Reading Vulkan 1.1 core features
Reading Vulkan 1.2 core features
Reading Vulkan 1.3 core features
Reading profiles
Reading Vulkan 1.1 core properties
Reading Vulkan 1.2 core properties
Reading Vulkan 1.3 core properties
Reading Vulkan 1.1 core features
Reading Vulkan 1.2 core features
Reading Vulkan 1.3 core features
Reading profiles

Thread 1 "vulkanCapsViewe" received signal SIGSEGV, Segmentation fault.
0x00005555555c13d0 in detail::VP_LUNARG_DESKTOP_PORTABILITY_2021::formatDesc::{lambda(VkBaseOutStructure*)#2}::_FUN(VkBaseOutStructure) [clone .lto_priv.3] [clone .lto_priv.0] ()
(gdb) bt
#0  0x00005555555c13d0 in detail::VP_LUNARG_DESKTOP_PORTABILITY_2021::formatDesc::{lambda(VkBaseOutStructure*)#2}::_FUN(VkBaseOutStructure) [clone .lto_priv.3] [clone .lto_priv.0] ()
#1  0x00005555555c7d4c in vpGetPhysicalDeviceProfileSupport(VkInstance_T*, VkPhysicalDevice_T*, VpProfileProperties const*, unsigned int*)::{lambda(VkBaseOutStructure*, void*)#3}::_FUN(VkBaseOutStructure*, void*) ()
#2  0x00005555555c2dea in detail::VP_LUNARG_DESKTOP_PORTABILITY_2021::chainerDesc::{lambda(VkBaseOutStructure*, void*, void (*)(VkBaseOutStructure, VkBaseOutStructure*))#4}::_FUN(VkBaseOutStructure, VkBaseOutStructure*, void (VkBaseOutStructure, VkBaseOutStructure*)) [clone .lto_priv.3] [clone .lto_priv.0] ()
#3  0x00005555555e5ed2 in vpGetPhysicalDeviceProfileSupport(VkInstance_T*, VkPhysicalDevice_T*, VpProfileProperties const*, unsigned int*) [clone .constprop.0] [clone .isra.0] ()
#4  0x00005555555ab7a1 in VulkanCapsViewer::getGPUinfo(VulkanDeviceInfo*, unsigned int, VkPhysicalDevice_T*) ()
#5  0x0000555555577406 in main ()
(gdb)
```

Manjaro, X11, Plasma KDE 5.24.4, Mesa 22.2 git RADV

**Comment by Krteq** [comment]
Issue was on my side

**Comment by Krteq** [comment]
Reopening issue

I downgraded mesa drivers to 22.0.2 and VulkanCapsViewer worked again, so there has to be some conflict between mesa-git RADV and newer VulkanCapsViewer versions

**Comment by SaschaWillems** [comment]
Judging by your trace above, it seems to happen while checking format support using the profiles library. If that's the case, then I probably can't do much about it.

**Comment by Krteq** [comment]
Thanks for checking trace.

So, is it possible there is some issue in RADV?

I checked another issue #143 opened today and it's also related to mesa-git

**Comment by SaschaWillems** [comment]
Yes, it may be possible. To test this would require a stand-alone app using the profiles library from https://github.com/KhronosGroup/Vulkan-Profiles. If that also crashes, it's most probably something in the driver or the library.

**Comment by SaschaWillems** [comment]
What you could do: Build the caps viewer and comment out the profiles part and export a JSON. If you attach it, I could check it for potential problems.

**Comment by SaschaWillems** [comment]
Can you test with one of these (depending on your platform):

* https://vulkan.gpuinfo.org/downloads/automated/Vulkan_Caps_Viewer-X11-x86_64.AppImage
* https://vulkan.gpuinfo.org/downloads/automated/Vulkan_Caps_Viewer-wayland-x86_64.AppImage

I have updated the profiles library to the latest version, and maybe this fixes your problem. If not we need to forward this problem to LunarG.

**Comment by codecnotsupported** [comment]
> Can you test with one of these (depending on your platform):
> 
>     * https://vulkan.gpuinfo.org/downloads/automated/Vulkan_Caps_Viewer-X11-x86_64.AppImage
> 
>     * https://vulkan.gpuinfo.org/downloads/automated/Vulkan_Caps_Viewer-wayland-x86_64.AppImage
> 
> 
> I have updated the profiles library to the latest version, and maybe this fixes your problem. If not we need to forward this problem to LunarG.

I tested both. Both segfault.
```
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
Reading Vulkan 1.1 core properties
Reading Vulkan 1.2 core properties
Reading Vulkan 1.3 core properties
Reading Vulkan 1.1 core features
Reading Vulkan 1.2 core features
Reading Vulkan 1.3 core features
Reading profiles
Reading profile  VP_KHR_roadmap_2022
Reading profile  VP_LUNARG_desktop_portability_2021
Reading profile  VP_LUNARG_desktop_portability_2021_subset
Reading Vulkan 1.1 core properties
Reading Vulkan 1.2 core properties
Reading Vulkan 1.3 core properties
Reading Vulkan 1.1 core features
Reading Vulkan 1.2 core features
Reading Vulkan 1.3 core features
Reading profiles
Reading profile  VP_KHR_roadmap_2022
Reading profile  VP_LUNARG_desktop_portability_2021
Segmentation fault (core dumped)
```

**Comment by dpanter** [comment]
> Can you test with one of these (depending on your platform):

Segfault here as well, coming from issue 143.

**Comment by SaschaWillems** [comment]
Thanks for testing. That confirms my initial suspicion. It looks like this is caused by the Profiles library and not my code and would require a fix from LunarG.

**Comment by dpanter** [comment]
Make a bugreport to [mesa](https://gitlab.freedesktop.org/mesa/mesa/-/issues) ?

**Comment by SaschaWillems** [comment]
I can't tell who is at fault. If you feel it's a mesa problem, feel free to open up an issue there :)

**Comment by dpanter** [comment]
https://gitlab.freedesktop.org/mesa/mesa/-/issues/6451

**Comment by Krteq** [comment]
Done and it seems you are right

Last working VulkanCapsViewer version is 3.11 for me and it shows 2 devices:
GPU0: AMD RADV VEGA10
GPU1: llvmpipe (LLVM 13.0.1, 256 bits)

**Comment by SaschaWillems** [comment]
Thanks for confirming. Hopefully this will help fix the issue.

**Comment by dpanter** [comment]
Closed #143 and my Mesa bugreport as vulkancapsviewer now works with mesa-git 22.3.

**Comment by Krteq** [comment]
I can confirm, issue is fixed now and VulkanCapsViewer is working even with llvmpipe
